lib/puppet/parser/functions/create_resources.rb

```diff
@@ -1,5 +1,10 @@
 Puppet::Parser::Functions::newfunction(:create_resources, :arity => -3, :doc => <<-'ENDHEREDOC') do |args|
     Converts a hash into a set of resources and adds them to the catalog.
+    
+    **Note**: Use this function selectively. It's generally better to write resources in
+     [Puppet](https://puppet.com/docs/puppet/latest/lang_resources.html) or with the
+      [Resource API](https://puppet.com/docs/puppet/6.17/create_types_and_providers_resource_api.html). 
```

Comment:
I would take this line about the Resource API out. It solves an adjacent problem, but not the same.